### PR TITLE
(fix) O3-3896: Tutorials modal should use a standard Carbon modal component

### PR DESCRIPTION
**Pull request by IamMujuziMoses**
## Requirements
- [x] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [x]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [x] My work includes tests or is validated by existing tests.

## Summary

This PR implements a fix for the Tutorial Modal styling of the [Carbon Modal](https://react.carbondesignsystem.com/?path=/docs/components-composedmodal--overview) component.

### Before
https://github.com/user-attachments/assets/0c6c0524-0ff4-44ce-89b4-8baa9935141c

### After
https://github.com/user-attachments/assets/4b5c8b76-5c36-4f2f-84ad-a232b288d212

## Related Issue

https://issues.openmrs.org/browse/O3-3896

### CODE REVIEW

**Comment by IamMujuziMoses** [comment]
@chibongho @vasharma05 @usamaidrsk could you please review this PR, your feedback will be appreciated.

**Comment by chibongho** [comment]
Thanks @IamMujuziMoses. Looks fine to me, but I'm not too familiar with this code base and don't have permission to approve PRs in this repo.

@jayasanka-sack @Vijaykv5 Can you take a look?

File: src/tutorial/modal.component.tsx [hunk 1]
```diff
@@ -39,20 +40,21 @@ const TutorialModal = ({ open, onClose }) => {
...
   return (
     <React.Fragment>
-      <ModalHeader closeModal={onClose} title={t('tutorial', 'Tutorial')}>
+      <ModalHeader className={styles.modalHeader} closeModal={onClose} title={t('tutorial', 'Tutorial')}>
         <p className={styles.description}>
           {t('modalDescription', 'Find walkthroughs and video tutorials on some of the core features of OpenMRS.')}
         </p>
       </ModalHeader>
-      <ModalBody className={styles.tutorialModal}>
+      <ModalBody className={styles.tutorialModal} hasScrollingContent >
```
Comment (Vijaykv5) [inline src/tutorial/modal.component.tsx:48]: Can we remove this? by default wrapping up over `<React.Fragment>` make it scrollable.
You can read it here [#docs ](https://o3-docs.openmrs.org/docs/modal-system)
```suggestion
      <ModalBody className={styles.tutorialModal} >
```

File: src/tutorial/modal.component.tsx [hunk 2]
```diff
@@ -39,20 +40,21 @@ const TutorialModal = ({ open, onClose }) => {
...
         <ul>
           {tutorials.map((tutorial, index) => (
             <li className={styles.tutorialItem} key={index}>
               <h3 className={styles.tutorialTitle}>{tutorial.title}</h3>
               <p className={styles.tutorialDescription}>{tutorial.description}</p>
-              <Button kind="ghost" onClick={() => handleWalkthroughClick(index)}>
+              <Link className={styles.tutorialLink} onClick={() => handleWalkthroughClick(index)} renderIcon={() => 
+                <ArrowRight aria-label="Arrow Right" />}>
```
Comment (Vijaykv5) [inline src/tutorial/modal.component.tsx:55]: @jayasanka-sack do we need to keep an `Arrow Right` here?
It's not defined in the design though.

@IamMujuziMoses Also please make the cursor to pointer

Comment (jayasanka-sack) [reply src/tutorial/modal.component.tsx:55]: I think the cursor is set to pointer by default for the link component. Please confirm.

**Review by jayasanka-sack — APPROVED** [review]
Thanks @IamMujuziMoses !

**Comment by jayasanka-sack** [comment]
The E2E test is failing when trying to click the walkthrough button. Could you please fix it? 
https://github.com/openmrs/openmrs-esm-user-onboarding/blob/9f24ae4ee0fb3c4a64c8c4e9ceacbdce842a1b5b/e2e/specs/onboarding-test.spec.ts#L28

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]
Please fix the failing E2E test as mentioned in here: https://github.com/openmrs/openmrs-esm-user-onboarding/pull/20#issuecomment-2388378291

**Pull request merged**